frontend/src/app/components/auth/auth.service.ts

```diff
@@ -155,7 +155,7 @@ export class AuthService {
   }
 
   isAuthenticated(): boolean {
-    return this._cookie.get('token') !== null;
+    return this._cookie.check('token') && this._cookie.get('token') !== null;
```

Comment:
**Suggestion:** You are checking the 'token' cookie twice in the `isAuthenticated()` method. You can simplify this by only checking if the cookie value is truthy, which will cover both the existence and non-nullity of the cookie. [performance]
```suggestion
    isAuthenticated(): boolean {
        return Boolean(this._cookie.get('token'));
    }
```